### PR TITLE
fix: [IDP-904]: Fix for checking if env variable present in config is…

### PR DESCRIPTION
**Pull request by devesh-harness**
… configured as well

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51240)

### CODE REVIEW

**Comment by devesh-harness** [comment]
trigger messagecheck

**Comment by devesh-harness** [comment]
trigger uts

File: idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java [hunk 1]
```diff
@@ -115,6 +124,35 @@ public List<String> getAllEnvVariablesForAccountIdentifierAndPluginId(String acc
...
+    if (!notConfiguredEnvVariables.isEmpty()) {
+      throw new UnsupportedOperationException(
+          String.format(ENV_VARIABLE_NOT_CONFIGURED_ERROR_MESSAGE, String.join(", ", notConfiguredEnvVariables)));
+    }
+  }
+
+  @VisibleForTesting
+  Set<String> getAllEnvVariableNamesFromBackstageEnvSecretVariable(
+      List<BackstageEnvSecretVariable> backstageEnvSecretVariables) {
+    return backstageEnvSecretVariables.stream().map(entity -> entity.getEnvName()).collect(Collectors.toSet());
```
Comment (vikyathharekal) [inline idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java:153]: nit: could use Lamda `BackstageEnvVariable::getEnvName`

File: idp-service/src/test/java/io/harness/idp/configmanager/resources/AppConfigApiImplTest.java [hunk 2]
```diff
@@ -44,6 +48,8 @@ public class AppConfigApiImplTest extends CategoryTest {
 
   @Mock ConfigManagerService configManagerService;
 
+  @Mock ConfigEnvVariablesService configEnvVariablesService;
```
Comment (vikyathharekal) [inline idp-service/src/test/java/io/harness/idp/configmanager/resources/AppConfigApiImplTest.java:51]: nit: unused

**Review by vikyathharekal — APPROVED** [review]

File: idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java (hunk 1 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java:153]: thanks for the suggestion will take this in the next PR as this is a small change.

File: idp-service/src/test/java/io/harness/idp/configmanager/resources/AppConfigApiImplTest.java (hunk 2 above)
Comment (devesh-harness) [reply idp-service/src/test/java/io/harness/idp/configmanager/resources/AppConfigApiImplTest.java:51]: as discussed this is needed for working of test case.

File: idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java [hunk 3]
```diff
@@ -115,6 +124,35 @@ public List<String> getAllEnvVariablesForAccountIdentifierAndPluginId(String acc
...
+    while (matcher.find()) {
+      fetchedEnvVariablesFromConfig.add(matcher.group(1));
+    }
+    Set<String> configuredEnvVariables =
+        getAllEnvVariableNamesFromBackstageEnvSecretVariable(appConfig.getEnvVariables());
+
+    Set<String> notConfiguredEnvVariables = Sets.difference(fetchedEnvVariablesFromConfig, configuredEnvVariables);
+
+    if (!notConfiguredEnvVariables.isEmpty()) {
+      throw new UnsupportedOperationException(
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java:145]: nit: InvalidRequestException is more suitable here

**Review by sathish-soundarapandian — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger review

**Pull request merged**